### PR TITLE
reference: update the transaction-related docs

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What is changed, added or deleted? 
Update the transaction-related docs.

### Which version does your change affect? 
dev/v3.1/v3.0/v2.1

### CODE REVIEW

File: dev/reference/transactions/transaction-isolation.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ TiDB 实现了快照隔离 (Snapshot Isolation) 级别的一致性。为与 MySQ
 
 > **注意：**
 >
-> 在 3.0 默认设置中，事务的自动重试已经默认关闭。关于该项功能的补充信息和应如何开启该项功能，请参考[自动重试导致的事务异常](#自动重试导致的事务异常)和[事务重试](#事务重试)。
+> 在 3.0 默认设置中，事务的自动重试已经默认关闭。关于该项功能对于隔离级别的影响和应如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
> 在 3.0 默认设置中，事务的自动重试功能已经默认关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```

File: dev/reference/transactions/transaction-isolation.md [hunk 2]
```diff
@@ -51,13 +51,11 @@ commit;                         |
 
 MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见，也就是说，即使该行在事务启动后被更新过，同样可以继续更新。这种情况在 TiDB 会导致事务回滚，导致事务最终失败，而 MySQL 是可以更新成功的。MySQL 的可重复读隔离级别并非 Snapshot 隔离级别，MySQL 可重复读隔离级别的一致性要弱于 Snapshot 隔离级别，也弱于 TiDB 的可重复读隔离级别。
 
-## 事务重试
+## 事务自动重试及带来的异常
 
-TiDB 默认不会进行事务重试，因为重试事务可能会导致更新丢失。如果业务可以容忍事务重试导致的异常，或者说并不关注事务是否以 SI 隔离级别来执行，可以开启自动重试。通过设置 `tidb_disable_txn_auto_retry = off` 可开启该项功能，同时要注意 `tidb_retry_limit` 的值不能为 0，否则，也会禁用自动重试。开启自动重试以后，事务遇到提交出错的可能性会降低。
+TiDB 默认不会进行事务重试，因为重试事务可能会导致更新丢失，从而破坏快照隔离。如果业务可以容忍事务重试导致的异常，或者说并不关注事务是否以快照隔离级别来执行，可以开启自动重试。通过设置 `tidb_disable_txn_auto_retry = off` 可开启该项功能，同时要注意 `tidb_retry_limit` 的值不能为 0，否则，也会禁用自动重试。开启自动重试以后，事务遇到提交出错的可能性会降低。
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
TiDB 默认不进行事务重试，因为重试事务可能会导致更新丢失，从而破坏快照隔离。如果业务可以容忍事务重试导致的异常，或并不关注事务是否以快照隔离级别来执行，则可以开启自动重试。通过设置 `tidb_disable_txn_auto_retry = off` 可开启该项功能。需注意 `tidb_retry_limit` 的值不能为 `0`，否则会禁用自动重试。开启自动重试以后，事务遇到提交出错的可能性会降低。
```

File: dev/reference/transactions/transaction-isolation.md [hunk 3]
```diff
@@ -51,13 +51,11 @@ commit;                         |
...
 
-## 自动重试导致的事务异常
-
-因为 TiDB 不会[默认](#事务重试)自动重试事务，如果开启自动重试，显式的事务在遇到冲突后，可能会导致最终结果不符合预期。
+因为 TiDB 不会自动重试事务，如果开启自动重试，显式的事务在遇到冲突后，可能会导致最终结果不符合预期。
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
TiDB 默认不进行自动重试事务。如果开启自动重试，显式的事务遇到冲突可能会导致最终结果不符合预期。
```

File: dev/reference/transactions/transaction-model.md [hunk 4]
```diff
@@ -47,30 +41,6 @@ UPDATE my_table SET a='newest_value' WHERE id = 3;
...
-* 事务的处理：
-
-    TiDB 在执行 `LOAD DATA` 操作时，默认将每 2 万行记录作为一个事务进行持久化存储。如果一次 `LOAD DATA` 操作插入的数据超过 2 万行，那么会分为多个事务进行提交。如果某个事务出错，这个事务会提交失败，但它前面的事务仍然会提交成功，在这种情况下，一次 `LOAD DATA` 操作会有部分数据插入成功，部分数据插入失败。而 MySQL 中会将一次 `LOAD DATA` 操作视为一个事务，如果其中发生失败情况，将会导致整个 `LOAD DATA` 操作失败。
-
-    > **注意：**
-    >
-    > `LOAD DATA` 在 TiDB 中默认会拆分事务后分批提交，这种拆分事务提交的方式是以打破事务的原子性和隔离性为代价的，使用该特性时，使用者需要保证没有其他对正在处理的表的**任何**操作，并且在出现报错时，需要及时**人工介入，检查数据的一致性和完整性**。不建议在生产环境中使用。
+> **注意：**
+>
+> 由于 TiDB 中的资源是分布式的，TiDB 中单线程 workload 可能不会很好的利用分布式资源，因此，性能相比于单实例部署的 MySQL 较低。这与 TiDB 中的事务延迟较高的情況类似。
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-model.md]:
```suggestion
> 由于 TiDB 中的资源是分布式的，TiDB 中单线程 workload 可能不会很好地利用分布式资源，因此性能相比于单实例部署的 MySQL 较低。这与 TiDB 中的事务延迟较高的情況类似。
```

File: v2.1/reference/transactions/transaction-model.md [hunk 5]
```diff
@@ -5,63 +5,42 @@ category: reference
...
 * 每个键值对不超过 6MB
-* 键值对的总数不超过 300,000
 * 键值对的总大小不超过 100MB
 
-## 小事务
+## 基于事务模型的优化实践
+
+由于 TiDB 中的每个事务都需要跟 PD leader 进行两次 round trip，TiDB 中的事务相比于 MySQL 中的事务延迟更高。以如下的 query 为例，用显式事务代替 `auto_commit`，可优化该 query 的性能。
+
+使用 auto_commit 的原始版本：
```
Comment (TomShawn) [inline v2.1/reference/transactions/transaction-model.md]:
```suggestion
使用 `auto_commit` 的原始版本：
```

File: dev/reference/transactions/transaction-model.md [hunk 6]
```diff
@@ -5,25 +5,19 @@ category: reference
...
 * 每个键值对不超过 6MB
 * 键值对的总大小不超过 100MB
 
-## 小事务
+## 基于事务模型的优化实践
 
-由于 TiDB 中的每个事务都需要跟 PD leader 进行两次 round trip，TiDB 中的小事务相比于 MySQL 中的小事务延迟更高。以如下的 query 为例，用显式事务代替 `auto_commit`，可优化该 query 的性能。
+由于 TiDB 中的每个事务都需要跟 PD leader 进行两次 round trip，TiDB 中的事务相比于 MySQL 中的事务延迟更高。以如下的 query 为例，用显式事务代替 `auto_commit`，可优化该 query 的性能。
 
 使用 auto_commit 的原始版本：
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-model.md]:
```suggestion
使用 `auto_commit` 的原始版本：
```

File: dev/reference/transactions/overview.md [hunk 7]
```diff
@@ -124,3 +125,31 @@ SELECT * FROM T; -- MySQL 返回 1 2；TiDB 返回 1
...
+
+```sql
+begin;
+insert into test values (1);
+insert into tset values (2);  -- tset 拼写错了，这条语句出错。
+insert into test values (3);
+commit;
+```
+
+上面的例子里面，第二个语句失败，其它插入 1 和 3 仍然能正常提交。
```
[the remark this answers is not in the document]
Comment (jackysp) [reply dev/reference/transactions/overview.md]: 1 and 3 here are the values inserted by the insert statements.

File: dev/reference/transactions/overview.md [hunk 8]
```diff
@@ -1,95 +1,97 @@
...
 
 {{< copyable "sql" >}}
 
 ```sql
-COMMIT;
+SET autocommit = {0 | 1}
 ```
 
-提交当前事务，包括从 `[BEGIN|START TRANSACTION]` 到 `COMMIT` 之间的所有修改。
+ 当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。 设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```
Comment (TomShawn) [inline dev/reference/transactions/overview.md]:
```suggestion
 当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```

File: dev/reference/transactions/overview.md [hunk 9]
```diff
@@ -1,95 +1,97 @@
...
 
-## ROLLBACK
+自动提交状态下，每条语句运行后，TiDB 会自动将修改提交到数据库中。非自动提交状态下，通过执行 `COMMIT` 语句来手动提交事务。注意某些语句执行后会导致隐式提交，详见[implicit commit](https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html)。例如，执行 `[BEGIN|START TRANCATION]` 语句的时候，TiDB 会试图提交上一个事务，并开启一个新的事务。
```
Comment (TomShawn) [inline dev/reference/transactions/overview.md]:
```suggestion
自动提交状态下，每条语句运行后，TiDB 会自动将修改提交到数据库中。非自动提交状态下，通过执行 `COMMIT` 语句来手动提交事务。注意某些语句执行后会导致隐式提交，详见 [implicit commit](https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html)。例如，执行 `[BEGIN|START TRANCATION]` 语句的时候，TiDB 会试图提交上一个事务，并开启一个新的事务。
```

File: dev/reference/transactions/overview.md [hunk 10]
```diff
@@ -1,95 +1,97 @@
...
+{{< copyable "sql" >}}
+
+```sql
+SET @@GLOBAL.autocommit = {0 | 1};
+```
 
 ## 显式事务和隐式事务
 
-TiDB 可以显式地使用事务（`[BEGIN|START TRANSACTION]`/`COMMIT`）或者隐式的使用事务（`SET autocommit = 1`）。
+TiDB 可以显式地使用事务（`[BEGIN|START TRANSACTION]`/`COMMIT`）或者隐式地使用事务（`SET autocommit = 1`）。
```
Comment (TomShawn) [inline dev/reference/transactions/overview.md]:
```suggestion
TiDB 可以显式地使用事务 (`[BEGIN|START TRANSACTION]`/`COMMIT`) 或者隐式地使用事务 (`SET autocommit = 1`)。
```

**Comment by jackysp** [comment]
But they're not exactly the same. @TomShawn

**Comment by jackysp** [comment]
PTAL @anotherrachel

File: dev/reference/transactions/transaction-isolation.md [hunk 11]
```diff
@@ -51,13 +51,11 @@ commit;                         |
 
 MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见，也就是说，即使该行在事务启动后被更新过，同样可以继续更新。这种情况在 TiDB 会导致事务回滚，导致事务最终失败，而 MySQL 是可以更新成功的。MySQL 的可重复读隔离级别并非 Snapshot 隔离级别，MySQL 可重复读隔离级别的一致性要弱于 Snapshot 隔离级别，也弱于 TiDB 的可重复读隔离级别。
 
-## 事务重试
+## 事务自动重试及带来的异常
```
[the remark this answers is not in the document]
Comment (jackysp) [reply dev/reference/transactions/transaction-isolation.md]: done

**Comment by jackysp** [comment]
All the modifications applied to v2.1, v3.0, and v3.1. PTAL @anotherrachel @TomShawn

File: dev/reference/transactions/overview.md [hunk 12]
```diff
@@ -1,95 +1,103 @@
...
+SET @@GLOBAL.autocommit = {0 | 1};
+```
 
 ## 显式事务和隐式事务
 
-TiDB 可以显式地使用事务（`[BEGIN|START TRANSACTION]`/`COMMIT`）或者隐式的使用事务（`SET autocommit = 1`）。
+TiDB 可以显式地使用事务 (`[BEGIN|START TRANSACTION]`/`COMMIT`) 或者隐式地使用事务 (`SET autocommit = 1`)。
 
-如果在 `autocommit = 1` 的状态下，通过 `[BEGIN|START TRANSACTION]` 语句开启一个新的事务，那么在 `COMMIT`/`ROLLBACK` 之前，会禁用 autocommit，也就是变成显式事务。
+自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句始终显式开启一个事务。当前会禁用自动提交，即隐式事务变成显式事务，直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到默认的自动提交状态。
```
Comment (TomShawn) [inline dev/reference/transactions/overview.md]:
```suggestion
在自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句会显式地开启一个事务，同时也会禁用自动提交，使隐式事务变成显式事务。直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到此前默认的自动提交状态。
```

File: dev/reference/transactions/overview.md [hunk 13]
```diff
@@ -1,95 +1,103 @@
...
 
-对于 DDL 语句，会自动提交并且不能回滚。如果运行 DDL 的时候，正在一个事务的中间过程中，会先将当前的事务提交，再执行 DDL。
+对于 DDL 语句，会自动提交并且不能回滚。如果运行 DDL 的时候，正在一个事务的中间过程中，会先自动提交当前事务，再执行 DDL。
```
Comment (TomShawn) [inline dev/reference/transactions/overview.md]: 对于 DDL 语句，隐式事务会自动提交语句且无法回滚？

File: dev/reference/transactions/transaction-isolation.md [hunk 14]
```diff
@@ -20,7 +20,7 @@ TiDB 实现了快照隔离 (Snapshot Isolation) 级别的一致性。为与 MySQ
 
 > **注意：**
 >
-> 在 3.0 默认设置中，事务的自动重试已经默认关闭。关于该项功能的补充信息和应如何开启该项功能，请参考[自动重试导致的事务异常](#自动重试导致的事务异常)和[事务重试](#事务重试)。
+> 在 3.0 默认设置中，事务的自动重试功能已经默认关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-isolation.md]:
```suggestion
> 在 TiDB v3.0 的默认设置中，事务的自动重试功能已经关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```

File: dev/reference/transactions/transaction-isolation.md [hunk 15]
```diff
@@ -51,13 +51,11 @@ commit;                         |
...
-## 事务重试
+## 事务自动重试及带来的异常
 
-TiDB 默认不会进行事务重试，因为重试事务可能会导致更新丢失。如果业务可以容忍事务重试导致的异常，或者说并不关注事务是否以 SI 隔离级别来执行，可以开启自动重试。通过设置 `tidb_disable_txn_auto_retry = off` 可开启该项功能，同时要注意 `tidb_retry_limit` 的值不能为 0，否则，也会禁用自动重试。开启自动重试以后，事务遇到提交出错的可能性会降低。
+TiDB 默认不进行事务重试，因为重试事务可能会导致更新丢失，从而破坏快照隔离。如果业务可以容忍事务重试导致的异常，或并不关注事务是否以快照隔离级别来执行，则可以开启自动重试。通过设置 `tidb_disable_txn_auto_retry = off` 可开启该项功能。需注意 `tidb_retry_limit` 的值不能为 `0`，否则会禁用自动重试。开启自动重试以后，事务遇到提交出错的可能性会降低。
 
-## 自动重试导致的事务异常
-
-因为 TiDB 不会[默认](#事务重试)自动重试事务，如果开启自动重试，显式的事务在遇到冲突后，可能会导致最终结果不符合预期。
+开启自动重试后，显式事务遇到冲突可能会导致最终结果不符合预期。比如下面这两个例子：
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-isolation.md]: 和下面一句重复了。
```suggestion
开启自动重试后，显式事务遇到冲突可能会导致最终结果不符合预期。
```

File: dev/reference/transactions/transaction-model.md [hunk 16]
```diff
@@ -47,30 +41,6 @@ UPDATE my_table SET a='newest_value' WHERE id = 3;
...
-* 事务的处理：
-
-    TiDB 在执行 `LOAD DATA` 操作时，默认将每 2 万行记录作为一个事务进行持久化存储。如果一次 `LOAD DATA` 操作插入的数据超过 2 万行，那么会分为多个事务进行提交。如果某个事务出错，这个事务会提交失败，但它前面的事务仍然会提交成功，在这种情况下，一次 `LOAD DATA` 操作会有部分数据插入成功，部分数据插入失败。而 MySQL 中会将一次 `LOAD DATA` 操作视为一个事务，如果其中发生失败情况，将会导致整个 `LOAD DATA` 操作失败。
-
-    > **注意：**
-    >
-    > `LOAD DATA` 在 TiDB 中默认会拆分事务后分批提交，这种拆分事务提交的方式是以打破事务的原子性和隔离性为代价的，使用该特性时，使用者需要保证没有其他对正在处理的表的**任何**操作，并且在出现报错时，需要及时**人工介入，检查数据的一致性和完整性**。不建议在生产环境中使用。
+> **注意：**
+>
+> 由于 TiDB 中的资源是分布式的，TiDB 中单线程 workload 可能不会很好地利用分布式资源，因此性能相比于单实例部署的 MySQL 较低。这与 TiDB 中的事务延迟较高的情況类似。
```
Comment (TomShawn) [inline dev/reference/transactions/transaction-model.md]:
```suggestion
> 由于 TiDB 中的资源是分布式的，TiDB 单线程 workload 中的分布式资源可能不会得到较好利用，因此性能相比于单实例部署的 MySQL 较低。这与 TiDB 中的事务延迟较高的情況类似。
```

Comment (jackysp) [reply dev/reference/transactions/transaction-model.md]: It changed the original mean.

File: dev/reference/transactions/overview.md (hunk 13 above)
Comment (jackysp) [reply dev/reference/transactions/overview.md]: yes

**Comment by jackysp** [comment]
Some changes suggest that I can't tell if it's just good, and I don't want to copy it in every version. @TomShawn

**Comment by TomShawn** [comment]
LGTM

File: dev/reference/transactions/overview.md [hunk 17]
```diff
@@ -1,95 +1,103 @@
...
 
 {{< copyable "sql" >}}
 
 ```sql
-COMMIT;
+SET autocommit = {0 | 1}
 ```
 
-提交当前事务，包括从 `[BEGIN|START TRANSACTION]` 到 `COMMIT` 之间的所有修改。
+ 当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```
Comment (anotherrachel) [inline dev/reference/transactions/overview.md]:
```suggestion
当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```

File: v2.1/reference/transactions/overview.md [hunk 18]
```diff
@@ -1,77 +1,110 @@
...
 
+{{< copyable "sql" >}}
+
 ```sql
-ROLLBACK;
+SET autocommit = {0 | 1}
 ```
 
-回滚当前事务，撤销从 `[BEGIN|START TRANSACTION]` 到 `ROLLBACK` 之间的所有修改。
+ 当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```
Comment (jackysp) [inline v2.1/reference/transactions/overview.md]:
```suggestion
当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```

File: v3.0/reference/transactions/overview.md [hunk 19]
```diff
@@ -1,96 +1,103 @@
...
 
 {{< copyable "sql" >}}
 
 ```sql
-COMMIT;
+SET autocommit = {0 | 1}
 ```
 
-提交当前事务，包括从 `[BEGIN|START TRANSACTION]` 到 `COMMIT` 之间的所有修改。
+ 当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```
Comment (jackysp) [inline v3.0/reference/transactions/overview.md]:
```suggestion
当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```

File: v3.1/reference/transactions/overview.md [hunk 20]
```diff
@@ -1,95 +1,103 @@
...
 
 {{< copyable "sql" >}}
 
 ```sql
-COMMIT;
+SET autocommit = {0 | 1}
 ```
 
-提交当前事务，包括从 `[BEGIN|START TRANSACTION]` 到 `COMMIT` 之间的所有修改。
+ 当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```
Comment (jackysp) [inline v3.1/reference/transactions/overview.md]:
```suggestion
当 `autocommit = 1` 时（默认），当前的 Session 为自动提交状态。设置 `autocommit = 0` 时将更改当前 Session 为非自动提交状态。
```

File: v2.1/reference/transactions/overview.md [hunk 21]
```diff
@@ -1,77 +1,110 @@
...
+SET @@GLOBAL.autocommit = {0 | 1};
+```
 
 ## 显式事务和隐式事务
 
-TiDB 可以显式地使用事务（`[BEGIN|START TRANSACTION]`/`COMMIT`）或者隐式的使用事务（`SET autocommit = 1`）。
+TiDB 可以显式地使用事务 (`[BEGIN|START TRANSACTION]`/`COMMIT`) 或者隐式地使用事务 (`SET autocommit = 1`)。
 
-如果在 `autocommit = 1` 的状态下，通过 `[BEGIN|START TRANSACTION]` 语句开启一个新的事务，那么在 `COMMIT`/`ROLLBACK` 之前，会禁用 autocommit，也就是变成显式事务。
+自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句始终显式开启一个事务。当前会禁用自动提交，即隐式事务变成显式事务，直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到默认的自动提交状态。
```
Comment (jackysp) [inline v2.1/reference/transactions/overview.md]:
```suggestion
在自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句会显式地开启一个事务，同时也会禁用自动提交，使隐式事务变成显式事务。直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到此前默认的自动提交状态。
```

File: v3.0/reference/transactions/overview.md [hunk 22]
```diff
@@ -1,96 +1,103 @@
...
+SET @@GLOBAL.autocommit = {0 | 1};
+```
 
 ## 显式事务和隐式事务
 
-TiDB 可以显式地使用事务（`[BEGIN|START TRANSACTION]`/`COMMIT`）或者隐式的使用事务（`SET autocommit = 1`）。
+TiDB 可以显式地使用事务 (`[BEGIN|START TRANSACTION]`/`COMMIT`) 或者隐式地使用事务 (`SET autocommit = 1`)。
 
-如果在 `autocommit = 1` 的状态下，通过 `[BEGIN|START TRANSACTION]` 语句开启一个新的事务，那么在 `COMMIT`/`ROLLBACK` 之前，会禁用 autocommit，也就是变成显式事务。
+自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句始终显式开启一个事务。当前会禁用自动提交，即隐式事务变成显式事务，直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到默认的自动提交状态。
```
Comment (jackysp) [inline v3.0/reference/transactions/overview.md]:
```suggestion
在自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句会显式地开启一个事务，同时也会禁用自动提交，使隐式事务变成显式事务。直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到此前默认的自动提交状态。
```

File: v3.1/reference/transactions/overview.md [hunk 23]
```diff
@@ -1,95 +1,103 @@
...
+SET @@GLOBAL.autocommit = {0 | 1};
+```
 
 ## 显式事务和隐式事务
 
-TiDB 可以显式地使用事务（`[BEGIN|START TRANSACTION]`/`COMMIT`）或者隐式的使用事务（`SET autocommit = 1`）。
+TiDB 可以显式地使用事务 (`[BEGIN|START TRANSACTION]`/`COMMIT`) 或者隐式地使用事务 (`SET autocommit = 1`)。
 
-如果在 `autocommit = 1` 的状态下，通过 `[BEGIN|START TRANSACTION]` 语句开启一个新的事务，那么在 `COMMIT`/`ROLLBACK` 之前，会禁用 autocommit，也就是变成显式事务。
+自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句始终显式开启一个事务。当前会禁用自动提交，即隐式事务变成显式事务，直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到默认的自动提交状态。
```
Comment (jackysp) [inline v3.1/reference/transactions/overview.md]:
```suggestion
在自动提交状态下，使用 `[BEGIN|START TRANSACTION]` 语句会显式地开启一个事务，同时也会禁用自动提交，使隐式事务变成显式事务。直到执行 `COMMIT` 或 `ROLLBACK` 语句时才会恢复到此前默认的自动提交状态。
```

File: v3.0/reference/transactions/transaction-isolation.md [hunk 24]
```diff
@@ -21,7 +20,7 @@ TiDB 实现了快照隔离 (Snapshot Isolation) 级别的一致性。为与 MySQ
 
 > **注意：**
 >
-> 在 3.0 默认设置中，事务的自动重试已经默认关闭。关于该项功能的补充信息和应如何开启该项功能，请参考[自动重试导致的事务异常](#自动重试导致的事务异常)和[事务重试](#事务重试)。
+> 在 3.0 默认设置中，事务的自动重试功能已经默认关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```
Comment (jackysp) [inline v3.0/reference/transactions/transaction-isolation.md]:
```suggestion
> 在 TiDB v3.0 的默认设置中，事务的自动重试功能已经关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```

File: v3.1/reference/transactions/transaction-isolation.md [hunk 25]
```diff
@@ -20,7 +20,7 @@ TiDB 实现了快照隔离 (Snapshot Isolation) 级别的一致性。为与 MySQ
 
 > **注意：**
 >
-> 在 3.0 默认设置中，事务的自动重试已经默认关闭。关于该项功能的补充信息和应如何开启该项功能，请参考[自动重试导致的事务异常](#自动重试导致的事务异常)和[事务重试](#事务重试)。
+> 在 3.0 默认设置中，事务的自动重试功能已经默认关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```
Comment (jackysp) [inline v3.1/reference/transactions/transaction-isolation.md]:
```suggestion
> 在 TiDB v3.0 的默认设置中，事务的自动重试功能已经关闭。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务自动重试及带来的异常](#事务自动重试及带来的异常)。
```

**Comment by jackysp** [comment]
PTAL @TomShawn @anotherrachel

**Pull request merged**